reportsrv/reportsrv.go

```diff
@@ -103,6 +109,19 @@ func (s *Server) serveReport(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	a, err := s.db.Analysis(s.db.NoTxn(ctx), aID)
+	if err != nil {
+		if strings.HasPrefix(string(aID), "analysis") {
+			s.logger.Error("failed to load analysis", zap.String("analysis_id", string(aID)), zap.Error(err))
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		} else {
+			s.logger.Info("poorly constructed analysis id", zap.String("path", string(aID)), zap.Error(err))
+			http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+			return
+		}
```

Comment:
Is this a real issue we expect to happen? Generally I don't like to treat the entity ID prefixes as "load-bearing", they're simply a useful debugging/sanity checking tool. What's wrong with using `db.IsErrNotFound`?

Reply:
Good question - I wanted to differentiate between them mostly for ease-of-debugging. There are some missing dependencies, and I'd like to distinguish in logs in such a way that we can alert about actual failures, and build in good filters for known missing deps.